### PR TITLE
CC-1381: Add anti-cheat test case for grep

### PR DESCRIPTION
**Pull request by andy1li**

### CODE REVIEW

File: internal/anti_cheat.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	input := "foobar"
+	pattern := "fo{1,2}bar"
+
+	executable.TimeoutInMilliseconds = 1000
+	result, err := executable.RunWithStdin([]byte(input), "-E", pattern)
+	if err != nil && err.Error() == "execution timed out" {
+		return nil
+	}
+
+	if result.ExitCode == cheatExitCode {
```
Comment (rohitpaulk) [inline internal/anti_cheat.go:23]: @andy1li I worry that we'll get false positives here. What if a program is implemented poorly such that it returns 0 on unknown inputs? 

I think we'll want to run two tests - one for exit code 1 and one for exit code 0, and make sure that BOTH of those pass. If that happens, it's safe to say that the implementation actually works (i.e. anti-cheat should trigger)

**Review by rohitpaulk — CHANGES_REQUESTED** [review]
Note added

File: internal/anti_cheat_quantifiers.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+func testQuantifiersAsAntiCheat(stageHarness *test_case_harness.TestCaseHarness) error {
+	testCases := []AntiCheatTestCase{
+		{
+			Pattern:          "fo{1,2}bar",
+			Input:            "foobar",
+			ExpectedExitCode: 0,
+		},
+		{
+			Pattern:          "cat",
```
Comment (rohitpaulk) [inline internal/anti_cheat_quantifiers.go:15]: @andy1li just reviving my previous comment here: 

> @andy1li I worry that we'll get false positives here. What if a program is implemented poorly such that it returns 0 on unknown inputs?

Even with this new test, wouldn't we still see problems? It's expected that almost all user implementations will handle a simple pattern correctly?

**Review by rohitpaulk — CHANGES_REQUESTED** [review]
Note added

File: internal/anti_cheat_quantifiers.go (hunk 2 above)
Comment (rohitpaulk) [reply internal/anti_cheat_quantifiers.go:15]: To recap the requirement here, we need a set of tests that: 

- (a) are unlikely to ALL pass against a regular user implementation
- (b) are very likely to ALL pass against an off-the-shelf regex implementation

Comment (andy1li) [reply internal/anti_cheat_quantifiers.go:15]: Got somewhat carried with all the code refactoring, but forgot the most essential test cases. 😭

Let me ponder this a bit more.

**Review by rohitpaulk — APPROVED** [review]

**Pull request merged**